iwyu_path_util.cc

```diff
@@ -149,6 +157,9 @@ string NormalizeFilePath(const string& path) {
   string result = CanonicalizeFilePath(path);
   while (StripLeft(&result, "./")) {
   }
+  // Remove trailing './' but don't split '../'
```

Comment:
Above all I want to keep it cohesive -- solve the smallest problem possible, with the right amount of redesign to make it fit well.

This one is borderline, I think, it would be nice if we could get a grip on paths, but maybe that's a separate change set.

I've always been confused by `Normalize` vs `Canonicalize`, I think it might be a case where someone just invented words to signify difference instead of profound meaning. And if one of them can be made redundant, I'm all for removing...